### ISSUE TITLE
Azure Automation Error calling another powershell using Start-AzAutomationRunbook

### ISSUE THREAD

**Issue by sgiovinetti** [Azure/azure-powershell#13350]
## Description
Running the command `Start-AzAutomationRunbook` to call another simple powershell retrieves the error:

```
Start-AzAutomationRunbook : A command that prompts the user failed because the host program or the command type does 

not support user interaction. The host was attempting to request confirmation with the following message: Object 

reference not set to an instance of an object.

```

## Steps to reproduce
The caller powershell
```powershell
Start-AzAutomationRunbook –AutomationAccountName 'xxx' –Name 'Test' -ResourceGroupName 'xxx' -Debug -Verbose
```
The target powershell (it can be also empty because in any case it is not called)
```powershell
Write-Output "Done"
```

## Environment data

```
Name                           Value                                                                                    

----                           -----                                                                                    

PSVersion                      5.1.15063.726                                                                            

PSEdition                      Desktop                                                                                  

PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                  

BuildVersion                   10.0.15063.726                                                                           

CLRVersion                     4.0.30319.42000                                                                          

WSManStackVersion              3.0                                                                                      

PSRemotingProtocolVersion      2.3                                                                                      

SerializationVersion           1.1.0.1                                                                                  

```

## Module versions

```powershell

    Directory: C:\Modules\User

ModuleType Version    Name                                ExportedCommands                                              

---------- -------    ----                                ----------------                                              

Script     1.9.5      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enabl...

Script     1.4.0      Az.Automation                       {Get-AzAutomationHybridWorkerGroup, Remove-AzAutomationHybr...

Script     2.2.1      Az.KeyVault                         {Add-AzKeyVaultCertificate, Update-AzKeyVaultCertificate, S...

Script     2.7.0      Az.Storage                          {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStora...

The runbook job failed due to a job stream being larger than 1MB, the limit that is supported by an Azure Automation sandbox. See some common ways to resolve this at https://aka.ms/AAjobstreamlimit
Script     6.1.1      AzureRM.Automation                  {Get-AzureRMAutomationHybridWorkerGroup, Remove-AzureRmAuto...

Script     5.9.1      AzureRM.Compute                     {Remove-AzureRmAvailabilitySet, Get-AzureRmAvailabilitySet,...

Script     5.8.3      AzureRM.Profile                     {Disable-AzureRmDataCollection, Disable-AzureRmContextAutos...

Script     6.7.3      AzureRM.Resources                   {Get-AzureRmProviderOperation, Remove-AzureRmRoleAssignment...

Script     4.12.1     AzureRM.Sql                         {Get-AzureRmSqlDatabaseTransparentDataEncryption, Get-Azure...

Script     5.2.0      AzureRM.Storage                     {Get-AzureRmStorageAccount, Get-AzureRmStorageAccountKey, N...

Manifest   1.0.867    MicrosoftPowerBIMgmt                                                                              

Binary     1.0.867    MicrosoftPowerBIMgmt.Admin          {Add-PowerBIEncryptionKey, Get-PowerBIEncryptionKey, Get-Po...

Binary     1.0.867    MicrosoftPowerBIMgmt.Capacities     Get-PowerBICapacity                                           

Binary     1.0.867    MicrosoftPowerBIMgmt.Data           {Add-PowerBIDataset, Set-PowerBITable, New-PowerBIDataset, ...

Binary     1.0.867    MicrosoftPowerBIMgmt.Profile        {Connect-PowerBIServiceAccount, Disconnect-PowerBIServiceAc...

Binary     1.0.867    MicrosoftPowerBIMgmt.Reports        {Get-PowerBIReport, New-PowerBIReport, Export-PowerBIReport...

Binary     1.0.867    MicrosoftPowerBIMgmt.Workspaces     {Get-PowerBIWorkspace, Get-PowerBIWorkspaceMigrationStatus,...

Script     21.1.18229 SqlServer                           {Add-RoleMember, Add-SqlAvailabilityDatabase, Add-SqlAvaila...

    Directory: C:\Modules\Global





ModuleType Version    Name                                ExportedCommands                                              

---------- -------    ----                                ----------------                                              

Manifest   1.1.0.0    AuditPolicyDsc                                                                                    

Manifest   1.0.3      Azure.Storage                       {Get-AzureStorageBlob, Get-AzureStorageBlobContent, Get-Azu...

Manifest   5.0.0.0    ComputerManagementDsc                                                                             

Script     0.2        GPRegistryPolicyParser              {Parse-PolFile, Read-RegistryPolicies, Create-RegistrySetti...

Script     0.0        Microsoft.PowerShell.Core                                                                         

Script     0.0        Microsoft.PowerShell.Diagnostics                                                                  

Script     0.0        Microsoft.PowerShell.Management                                                                   

Script     0.0        Microsoft.PowerShell.Security                                                                     

Script     0.0        Microsoft.PowerShell.Utility                                                                      

Script     0.0        Microsoft.WSMan.Management                                                                        

Binary     1.0        Orchestrator.AssetManagement.Cmd... {Get-AutomationCertificate, Get-AutomationConnection, Get-S...

Manifest   2.9.0.0    PSDscResources                                                                                    

Manifest   2.1.0.0    SecurityPolicyDsc                                                                                 

Manifest   1.0        StateConfigCompositeResources                                                                     

Manifest   1.1        xDSCDomainjoin                                                                                    

Manifest   1.1.0.0    xPowerShellExecutionPolicy          {Get-TargetResource, Test-TargetResource, Set-TargetResource} 

Manifest   1.1.0.0    xRemoteDesktopAdmin                                                                               

    Directory: C:\Program Files\WindowsPowerShell\Modules





ModuleType Version    Name                                ExportedCommands                                              

---------- -------    ----                                ----------------                                              

Script     1.0.1      Microsoft.PowerShell.Operation.V... {Get-OperationValidation, Invoke-OperationValidation}         

Binary     1.0.0.1    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-Packag...

Script     3.4.0      Pester                              {Describe, Context, It, Should...}                            

Script     1.0.0.1    PowerShellGet                       {Install-Module, Find-Module, Save-Module, Update-Module...}  

Script     1.2        PSReadline                          {Get-PSReadlineKeyHandler, Set-PSReadlineKeyHandler, Remove...

    Directory: C:\Windows\system32\WindowsPowerShell\v1.0\Modules





ModuleType Version    Name                                ExportedCommands                                              

---------- -------    ----                                ----------------                                              

Manifest   1.0.0.0    AppBackgroundTask                   {Disable-AppBackgroundTaskDiagnosticLog, Enable-AppBackgrou...

Manifest   2.0.0.0    Appx                                {Add-AppxPackage, Get-AppxPackage, Get-AppxPackageManifest,...

Manifest   2.0.0.0    BitsTransfer                        {Add-BitsFile, Complete-BitsTransfer, Get-BitsTransfer, Rem...

Manifest   1.0.0.0    CimCmdlets                          {Get-CimAssociatedInstance, Get-CimClass, Get-CimInstance, ...

Manifest   1.0        Defender                            {Get-MpPreference, Set-MpPreference, Add-MpPreference, Remo...

Manifest   1.0.0.0    DeliveryOptimization                {Get-DeliveryOptimizationStatus, Get-DeliveryOptimizationPe...

Manifest   1.0.0.0    DirectAccessClientComponents        {Disable-DAManualEntryPointSelection, Enable-DAManualEntryP...

Script     3.0        Dism                                {Add-AppxProvisionedPackage, Add-WindowsDriver, Add-Windows...

Manifest   1.0.0.0    DnsClient                           {Resolve-DnsName, Clear-DnsClientCache, Get-DnsClient, Get-...

Manifest   1.0.0.0    EventTracingManagement              {Start-EtwTraceSession, New-EtwTraceSession, Get-EtwTraceSe...

Manifest   2.0.0.0    International                       {Get-WinDefaultInputMethodOverride, Set-WinDefaultInputMeth...

Manifest   1.0.0.0    iSCSI                               {Get-IscsiTargetPortal, New-IscsiTargetPortal, Remove-Iscsi...

Script     1.0.0.0    ISE                                 {New-IseSnippet, Import-IseSnippet, Get-IseSnippet}           

Manifest   1.0.0.0    Kds                                 {Add-KdsRootKey, Get-KdsRootKey, Test-KdsRootKey, Set-KdsCo...

Manifest   1.0.1.0    Microsoft.PowerShell.Archive        {Compress-Archive, Expand-Archive}                            

Manifest   3.0.0.0    Microsoft.PowerShell.Diagnostics    {Get-WinEvent, Get-Counter, Import-Counter, Export-Counter...}

Manifest   3.0.0.0    Microsoft.PowerShell.Host           {Start-Transcript, Stop-Transcript}                           

Manifest   1.0.0.0    Microsoft.PowerShell.LocalAccounts  {Add-LocalGroupMember, Disable-LocalUser, Enable-LocalUser,...

Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-ItemProperty, Join-Path...}

Script     1.0        Microsoft.PowerShell.ODataUtils     Export-ODataEndpointProxy                                     

Manifest   3.0.0.0    Microsoft.PowerShell.Security       {Get-Acl, Set-Acl, Get-PfxCertificate, Get-Credential...}     

Manifest   3.1.0.0    Microsoft.PowerShell.Utility        {Format-List, Format-Custom, Format-Table, Format-Wide...}    

Manifest   3.0.0.0    Microsoft.WSMan.Management          {Disable-WSManCredSSP, Enable-WSManCredSSP, Get-WSManCredSS...

Manifest   1.0        MMAgent                             {Disable-MMAgent, Enable-MMAgent, Set-MMAgent, Get-MMAgent...}

Manifest   1.0.0.0    MsDtc                               {New-DtcDiagnosticTransaction, Complete-DtcDiagnosticTransa...

Manifest   2.0.0.0    NetAdapter                          {Disable-NetAdapter, Disable-NetAdapterBinding, Disable-Net...

Manifest   1.0.0.0    NetConnection                       {Get-NetConnectionProfile, Set-NetConnectionProfile}          

Manifest   1.0.0.0    NetEventPacketCapture               {New-NetEventSession, Remove-NetEventSession, Get-NetEventS...

Manifest   2.0.0.0    NetLbfo                             {Add-NetLbfoTeamMember, Add-NetLbfoTeamNic, Get-NetLbfoTeam...

Manifest   1.0.0.0    NetNat                              {Get-NetNat, Get-NetNatExternalAddress, Get-NetNatStaticMap...

Manifest   2.0.0.0    NetQos                              {Get-NetQosPolicy, Set-NetQosPolicy, Remove-NetQosPolicy, N...

Manifest   2.0.0.0    NetSecurity                         {Get-DAPolicyChange, New-NetIPsecAuthProposal, New-NetIPsec...

Manifest   1.0.0.0    NetSwitchTeam                       {New-NetSwitchTeam, Remove-NetSwitchTeam, Get-NetSwitchTeam...

Manifest   1.0.0.0    NetTCPIP                            {Get-NetIPAddress, Get-NetIPInterface, Get-NetIPv4Protocol,...

Manifest   1.0.0.0    NetworkConnectivityStatus           {Get-DAConnectionStatus, Get-NCSIPolicyConfiguration, Reset...

Manifest   1.0.0.0    NetworkSwitchManager                {Disable-NetworkSwitchEthernetPort, Enable-NetworkSwitchEth...

Manifest   1.0.0.0    NetworkTransition                   {Add-NetIPHttpsCertBinding, Disable-NetDnsTransitionConfigu...

Manifest   1.0.0.0    PcsvDevice                          {Get-PcsvDevice, Start-PcsvDevice, Stop-PcsvDevice, Restart...

Manifest   1.0.0.0    PKI                                 {Add-CertificateEnrollmentPolicyServer, Export-Certificate,...

Manifest   1.0.0.0    PnpDevice                           {Get-PnpDevice, Get-PnpDeviceProperty, Enable-PnpDevice, Di...

Manifest   1.1        PrintManagement                     {Add-Printer, Add-PrinterDriver, Add-PrinterPort, Get-Print...

Script     3.0        Provisioning                        {Install-ProvisioningPackage, Export-ProvisioningPackage, I...

Manifest   1.1        PSDesiredStateConfiguration         {Set-DscLocalConfigurationManager, Start-DscConfiguration, ...

Script     1.0.0.0    PSDiagnostics                       {Disable-PSTrace, Disable-PSWSManCombinedTrace, Disable-WSM...

Binary     1.1.0.0    PSScheduledJob                      {New-JobTrigger, Add-JobTrigger, Remove-JobTrigger, Get-Job...

Manifest   2.0.0.0    PSWorkflow                          {New-PSWorkflowExecutionOption, New-PSWorkflowSession, nwsn}  

Manifest   1.0.0.0    PSWorkflowUtility                   Invoke-AsWorkflow                                             

Manifest   1.0.0.0    ScheduledTasks                      {Get-ScheduledTask, Set-ScheduledTask, Register-ScheduledTa...

Manifest   2.0.0.0    SecureBoot                          {Confirm-SecureBootUEFI, Set-SecureBootUEFI, Get-SecureBoot...

Manifest   2.0.0.0    SmbShare                            {Get-SmbShare, Remove-SmbShare, Set-SmbShare, Block-SmbShar...

Manifest   2.0.0.0    SmbWitness                          {Get-SmbWitnessClient, Move-SmbWitnessClient, gsmbw, msmbw...}

Manifest   1.0.0.0    StartLayout                         {Export-StartLayout, Import-StartLayout, Export-StartLayout...

Manifest   2.0.0.0    Storage                             {Add-InitiatorIdToMaskingSet, Add-PartitionAccessPath, Add-...

Manifest   2.0.0.0    TLS                                 {New-TlsSessionTicketKey, Enable-TlsSessionTicketKey, Disab...

Manifest   1.0.0.0    TroubleshootingPack                 {Get-TroubleshootingPack, Invoke-TroubleshootingPack}         

Manifest   2.0.0.0    TrustedPlatformModule               {Get-Tpm, Initialize-Tpm, Clear-Tpm, Unblock-Tpm...}          

Manifest   2.0.0.0    VpnClient                           {Add-VpnConnection, Set-VpnConnection, Remove-VpnConnection...

Manifest   1.0.0.0    Wdac                                {Get-OdbcDriver, Set-OdbcDriver, Get-OdbcDsn, Add-OdbcDsn...} 

Manifest   1.0.0.0    WindowsDeveloperLicense             {Get-WindowsDeveloperLicense, Unregister-WindowsDeveloperLi...

Script     1.0        WindowsErrorReporting               {Enable-WindowsErrorReporting, Disable-WindowsErrorReportin...

Manifest   1.0.0.0    WindowsSearch                       {Get-WindowsSearchSetting, Set-WindowsSearchSetting}          

Manifest   1.0.0.0    WindowsUpdate                       Get-WindowsUpdateLog                                          

```

## Debug output

```
Sought all Az modules and got latest version 0.0.0

12:13:34 PM - StartAzureAutomationRunbook begin processing with ParameterSet 'ByAsynchronousReturnJob'.

Start-AzAutomationRunbook : A command that prompts the user failed because the host program or the command type does 

not support user interaction. The host was attempting to request confirmation with the following message: Object 

reference not set to an instance of an object.

At line:4 char:1

+ Start-AzAutomationRunbook –AutomationAccountName 'MACustomerAutomatio ...

+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

    + CategoryInfo          : NotImplemented: (:) [Start-AzAutomationRunbook], HostException

    + FullyQualifiedErrorId : 

HostFunctionNotImplemented,Microsoft.Azure.Commands.Automation.Cmdlet.StartAzureAutomationRunbook

```

## Error output

```


12:13:35 PM - ResolveError begin processing with ParameterSet 'AnyErrorParameterSet'.

Upcoming breaking changes in the cmdlet 'Resolve-AzError' :

The `Resolve-Error` alias will be removed in a future release.  Please change any scripts that use this alias to use 

`Resolve-AzError` instead.

Note : Go to https://aka.ms/azps-changewarnings for steps to suppress this breaking change warning, and other 

information on breaking changes in Azure PowerShell.



InnerException   : False

Exception        : System.Management.Automation.Host.HostException: A command that prompts the user failed because the 

                   host program or the command type does not support user interaction. The host was attempting to 

                   request confirmation with the following message: Object reference not set to an instance of an 

                   object.

                      at System.Management.Automation.Internal.Host.InternalHostUserInterface.ThrowPromptNotInteractive(

                   String promptMessage)

                      at System.Management.Automation.Internal.Host.InternalHostUserInterface.PromptForChoice(String 

                   caption, String message, Collection`1 choices, Int32 defaultChoice)

                      at System.Management.Automation.MshCommandRuntime.InquireHelper(String inquireMessage, String 

                   inquireCaption, Boolean allowYesToAll, Boolean allowNoToAll, Boolean replaceNoWithHalt, Boolean 

                   hasSecurityImpact)

                      at System.Management.Automation.MshCommandRuntime._WriteErrorSkipAllowCheck(ErrorRecord 

                   errorRecord, Nullable`1 actionPreference)

                      at System.Management.Automation.MshCommandRuntime.DoWriteError(Object obj)

                      at System.Security.SecurityContext.Run(SecurityContext securityContext, ContextCallback callback, 

                   Object state)

                      at System.Management.Automation.MshCommandRuntime.WriteError(ErrorRecord errorRecord, Boolean 

                   overrideInquire)

                      at System.Management.Automation.Cmdlet.WriteError(ErrorRecord errorRecord)

                      at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.WriteError(ErrorRecord 

                   errorRecord)

                      at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()

                      at System.Management.Automation.CommandProcessor.ProcessRecord()

Message          : A command that prompts the user failed because the host program or the command type does not support 

                   user interaction. The host was attempting to request confirmation with the following message: Object 

                   reference not set to an instance of an object.

StackTrace       :    at System.Management.Automation.Internal.Host.InternalHostUserInterface.ThrowPromptNotInteractive(

                   String promptMessage)

                      at System.Management.Automation.Internal.Host.InternalHostUserInterface.PromptForChoice(String 

                   caption, String message, Collection`1 choices, Int32 defaultChoice)

                      at System.Management.Automation.MshCommandRuntime.InquireHelper(String inquireMessage, String 

                   inquireCaption, Boolean allowYesToAll, Boolean allowNoToAll, Boolean replaceNoWithHalt, Boolean 

                   hasSecurityImpact)

                      at System.Management.Automation.MshCommandRuntime._WriteErrorSkipAllowCheck(ErrorRecord 

                   errorRecord, Nullable`1 actionPreference)

                      at System.Management.Automation.MshCommandRuntime.DoWriteError(Object obj)

                      at System.Security.SecurityContext.Run(SecurityContext securityContext, ContextCallback callback, 

                   Object state)

                      at System.Management.Automation.MshCommandRuntime.WriteError(ErrorRecord errorRecord, Boolean 

                   overrideInquire)

                      at System.Management.Automation.Cmdlet.WriteError(ErrorRecord errorRecord)

                      at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.WriteError(ErrorRecord 

                   errorRecord)

                      at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord()

                      at System.Management.Automation.CommandProcessor.ProcessRecord()

HelpLink         : 

ErrorDetails     : 

ErrorCategory    : NotImplemented: (:) [Start-AzAutomationRunbook], HostException

InvocationInfo   : System.Management.Automation.InvocationInfo

ScriptStackTrace : at <ScriptBlock>, <No file>: line 4

AzureQoSEvent: CommandName - Resolve-AzError; IsSuccess - True; Duration - 00:00:00.3003925;

Finish sending metric.

12:13:36 PM - ResolveError end processing.

```

**Comment by dingmeng-xue** [comment]
@sgiovinetti , thanks for reporting. Could you share the versions of Az modules in your automation account?

In addition, I will involve automation team.

**Comment by sgiovinetti** [comment]
@dingmeng-xue 
List and versions are in the bug i report again here:

```                                              
Script     1.9.5      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enabl...

Script     1.4.0      Az.Automation                       {Get-AzAutomationHybridWorkerGroup, Remove-AzAutomationHybr...

Script     2.2.1      Az.KeyVault                         {Add-AzKeyVaultCertificate, Update-AzKeyVaultCertificate, S...

Script     2.7.0      Az.Storage                          {Get-AzStorageAccount, Get-AzStorageAccountKey, New-AzStora...
```

**Comment by dingmeng-xue** [comment]
@sgiovinetti , I meant whether you can check module installed on your automation account. You can find that information via Azure Portal -> your automation account -> Shared Resources -> Modules. 

I created a same runbook but cannot reproduce your problem. I involve Automation team.

**Comment by jaspkaur28** [comment]
We are investigating the issue. Will share an update by next week .

**Comment by jaspkaur28** [comment]
We've got it added to our backlog as an improvement area & we are trying to prioritize it for Q1 - Feb , March timeframe.

**Comment by jaspkaur28** [comment]
Deferred to CY2021 Q4 due to other priorities.

**Comment by SatishBoddu-MSFT** [comment]
Hello @jaspkaur28 Could you please help us in this matter? Please advice us if twe good to close this issue?

Thanks for your help!

**Comment by jaspkaur28** [comment]
@SatishBoddu-MSFT The ETA for this issue is 11/30.
